### PR TITLE
Update broken links tia index 3372

### LINKED ISSUE #3372: Broken links detected in docs 🔗
# Summary

| Status        | Count |
|---------------|-------|
| 🔍 Total      | 1371  |
| ✅ Successful | 1353  |
| ⏳ Timeouts   | 0     |
| 🔀 Redirected | 0     |
| 👻 Excluded   | 9     |
| ❓ Unknown    | 0     |
| 🚫 Errors     | 7     |

### Errors in intermediate_source/tiatoolbox_tutorial.rst

* [404] [https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=pretrained%20models#tiatoolbox.models.architecture.get_pretrained_model](https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=pretrained%20models#tiatoolbox.models.architecture.get_pretrained_model) | Network error: Not Found
* [404] [https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=WSIReader.read_rect#tiatoolbox.wsicore.wsireader.WSIReader.read_rect](https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=WSIReader.read_rect#tiatoolbox.wsicore.wsireader.WSIReader.read_rect) | Network error: Not Found

### Errors in index.rst

* [404] [https://pytorch.org/executorch/stable/demo-apps-android.html](https://pytorch.org/executorch/stable/demo-apps-android.html) | Network error: Not Found
* [404] [https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html](https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html) | Network error: Not Found
* [404] [https://pytorch.org/executorch/stable/demo-apps-ios.html](https://pytorch.org/executorch/stable/demo-apps-ios.html) | Network error: Not Found

### Errors in recipes_source/deployment_with_flask.rst

* [ERROR] [http://localhost:5000/predict](http://localhost:5000/predict) | Network error: error sending request for url (http://localhost:5000/predict) Maybe a certificate error?

### Errors in intermediate_source/torchserve_with_ipex_2.rst

* [ERROR] [https://easyperf.net/blog/2019/02/09/Top-Down-performance-analysis-methodology](https://easyperf.net/blog/2019/02/09/Top-Down-performance-analysis-methodology) | Network error: error sending request for url (https://easyperf.net/blog/2019/02/09/Top-Down-performance-analysis-methodology) Maybe a certificate error?
[Full Github Actions output](https://github.com/pytorch/tutorials/actions/runs/15368908633?check_suite_focus=true)

cc @svekars @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by nirajkamal**
Fixes #3372

## Description
Errors in intermediate_source/tiatoolbox_tutorial.rst
[404] https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=pretrained%20models#tiatoolbox.models.architecture.get_pretrained_model | Network error: Not Found
[404] https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=WSIReader.read_rect#tiatoolbox.wsicore.wsireader.WSIReader.read_rect | Network error: Not Found
Errors in index.rst
[404] https://pytorch.org/executorch/stable/demo-apps-android.html | Network error: Not Found
[404] https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html | Network error: Not Found
[404] https://pytorch.org/executorch/stable/demo-apps-ios.html | Network error: Not Found

Changes:

Location | Old link [404] | Working Link
-- | -- | --
intermediate_source/tiatoolbox_tutorial.rst | https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=pretrained%20models#tiatoolbox.models.architecture.get_pretrained_model | https://tia-toolbox.readthedocs.io/en/stable/_autosummary/tiatoolbox.models.architecture.get_pretrained_model.html#tiatoolbox.models.architecture.get_pretrained_model
intermediate_source/tiatoolbox_tutorial.rst | https://tia-toolbox.readthedocs.io/en/latest/usage.html?highlight=WSIReader.read_rect#tiatoolbox.wsicore.wsireader.WSIReader.read_rect | https://tia-toolbox.readthedocs.io/en/stable/_autosummary/tiatoolbox.wsicore.wsireader.WSIReader.html#tiatoolbox.wsicore.wsireader.WSIReader.read_rect

Location | Old link [404] | Currently updated Working Link 
-- | -- | --  
index.rst | https://pytorch.org/executorch/stable/demo-apps-android.html | https://github.com/pytorch-labs/executorch-examples/tree/main/dl3/android/DeepLabV3Demo#executorch-android-demo-app 
index.rst | https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html | https://docs.pytorch.org/executorch/main/tutorials/devtools-integration-tutorial.html |  
index.rst | https://pytorch.org/executorch/stable/demo-apps-ios.html | https://github.com/pytorch-labs/executorch-examples/tree/main/mv3/apple/ExecuTorchDemo 

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #3372")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

cc - @spzala @svekars

### CODE REVIEW

**Comment by nirajkamal** [comment]
The GitHub links are valid - but link checking fails maybe due to capcha. 

Need suggestions weather to use links from 0.5 docs version instead. I have however seen the same GitHub links used in other parts of the documentation and hence used it in this PR.


Location | Old link [404] | Currently updated Working Link | Working link from version 0.5
-- | -- | -- | --
index.rst | https://pytorch.org/executorch/stable/demo-apps-android.html | (github) https://github.com/pytorch-labs/executorch-examples/tree/main/dl3/android/DeepLabV3Demo#executorch-android-demo-app | https://docs.pytorch.org/executorch/0.5/demo-apps-android.html
index.rst | https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html | https://docs.pytorch.org/executorch/main/tutorials/devtools-integration-tutorial.html |  
index.rst | https://pytorch.org/executorch/stable/demo-apps-ios.html | (github) https://github.com/pytorch-labs/executorch-examples/tree/main/mv3/apple/ExecuTorchDemo | https://docs.pytorch.org/executorch/0.5/demo-apps-ios.html


cc - @spzala @svekars

**Comment by sekyondaMeta** [comment]
> The GitHub links are valid - but link checking fails maybe due to capcha.
> 
> Need suggestions weather to use links from 0.5 docs version instead. I have however seen the same GitHub links used in other parts of the documentation and hence used it in this PR.
> 
> Location	Old link [404]	Currently updated Working Link	Working link from version 0.5
> index.rst	https://pytorch.org/executorch/stable/demo-apps-android.html	(github) https://github.com/pytorch-labs/executorch-examples/tree/main/dl3/android/DeepLabV3Demo#executorch-android-demo-app	https://docs.pytorch.org/executorch/0.5/demo-apps-android.html
> index.rst	https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html	https://docs.pytorch.org/executorch/main/tutorials/devtools-integration-tutorial.html	 
> index.rst	https://pytorch.org/executorch/stable/demo-apps-ios.html	(github) https://github.com/pytorch-labs/executorch-examples/tree/main/mv3/apple/ExecuTorchDemo	https://docs.pytorch.org/executorch/0.5/demo-apps-ios.html
> cc - @spzala @svekars

Looks like the rest of the links are passing but one. Can you double check the one thats failing

**Comment by nirajkamal** [comment]
> > The GitHub links are valid - but link checking fails maybe due to capcha.
> > Need suggestions weather to use links from 0.5 docs version instead. I have however seen the same GitHub links used in other parts of the documentation and hence used it in this PR.
> > Location	Old link [404]	Currently updated Working Link	Working link from version 0.5
> > index.rst	https://pytorch.org/executorch/stable/demo-apps-android.html	(github) https://github.com/pytorch-labs/executorch-examples/tree/main/dl3/android/DeepLabV3Demo#executorch-android-demo-app	https://docs.pytorch.org/executorch/0.5/demo-apps-android.html
> > index.rst	https://pytorch.org/executorch/stable/tutorials/sdk-integration-tutorial.html	https://docs.pytorch.org/executorch/main/tutorials/devtools-integration-tutorial.html	 
> > index.rst	https://pytorch.org/executorch/stable/demo-apps-ios.html	(github) https://github.com/pytorch-labs/executorch-examples/tree/main/mv3/apple/ExecuTorchDemo	https://docs.pytorch.org/executorch/0.5/demo-apps-ios.html
> > cc - @spzala @svekars
> 
> Looks like the rest of the links are passing but one. Can you double check the one thats failing

Found the issue - https://github.com/pytorch-labs/executorch-examples/tree/main/mv3/apple/ExecuTorchDemo in index.rst
@sekyondaMeta This link now works. The previous link is all lower case, it works in chrome but failed in safari - fixed it.

**Comment by nirajkamal** [comment]
Fixed the GitHub link - it was due to case sensitivity while parsing the link in some browsers.

**Review by sekyondaMeta — APPROVED** [review]

**Comment by nirajkamal** [comment]
Thank you @sekyondaMeta
